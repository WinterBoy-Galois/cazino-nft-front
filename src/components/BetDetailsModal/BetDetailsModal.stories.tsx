import React from 'react';
import { storiesOf } from '@storybook/react';
import BetDetailsModal from '.';
import { GameTypes } from '../../models/gameTypes.model';

storiesOf('Components/BetDetailsModal', module).add('default', () => (
  <BetDetailsModal
    show={true}
    bet={{
      id: '278192',
      username: 'gutierrezbrian',
      time: 1591032136876,
      userid: 67,
      gameid: GameTypes.MINES,
      bet: 0.00009425,
      profit: 0.00002852,
      multiplier: 1.3026315789473684,
    }}
  />
));
