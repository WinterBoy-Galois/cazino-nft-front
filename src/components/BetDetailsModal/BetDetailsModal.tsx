import React, { useMemo } from 'react';
import PageableModal from '../PageableModal';
import Bet from '../../models/bet.model';
import { useTranslation } from 'react-i18next';
import { BetDetailsPageWithData } from './components/BetDetailsPage/BetDetailsPage';
import { BetResultsPageWithData } from './components/BetResultsPage/BetResultsPage';
import { ServerSeedPageWithData } from './components/ServerSeedPage/ServerSeedPage';
import { useLocation, useNavigate } from '@reach/router';

interface IProps {
  show: boolean;
  bet: Bet;
  onClose?: () => void;
  activePage?: number;
  confirmed?: boolean;
  cancelled?: boolean;
}

const BetDetailsModal: React.FC<IProps> = ({ show, onClose, bet, activePage }) => {
  const { t } = useTranslation(['modals']);
  const location = useLocation();
  const navigate = useNavigate();
  const pages = useMemo(
    () => [
      <BetDetailsPageWithData key={1} bet={bet} />,
      <BetResultsPageWithData key={2} gameType={bet?.gameid} betId={bet?.id} />,
      <ServerSeedPageWithData key={3} bet={bet} />,
    ],
    [bet]
  );

  if (!bet && !show) {
    return null;
  } else if (!bet) {
    navigate(location.pathname);
    return null;
  }

  return (
    <PageableModal
      show={show}
      title={[t('betDetails.title'), t('betResults.title'), t('serverSeed.title')]}
      onClose={onClose}
      pages={pages}
      activePage={activePage}
    />
  );
};

export default BetDetailsModal;
