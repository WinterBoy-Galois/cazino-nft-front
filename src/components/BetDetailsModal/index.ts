export { default } from './BetDetailsModal';
export { BetDetailsModalWithData } from './BetDetailsModal';
