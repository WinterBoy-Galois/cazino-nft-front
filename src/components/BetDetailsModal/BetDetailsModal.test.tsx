import React from 'react';
import { render } from '@testing-library/react';
import BetDetailsModal from '.';
import { LocationProvider } from '@reach/router';

describe('BetDetailsModal', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(
      <LocationProvider>
        <BetDetailsModal show={true} />
      </LocationProvider>
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
