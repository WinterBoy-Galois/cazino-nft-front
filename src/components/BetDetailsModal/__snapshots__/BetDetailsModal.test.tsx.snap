// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetDetailsModal should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="container"
      >
        <div
          aria-modal="true"
          class="modal scale--enter scale--enter-active"
          role="dialog"
        >
          <div
            class="modal__header"
          >
            <div
              class="modal__header__headline"
            >
              Bet Details
            </div>
            <div
              class="modal__header__close"
            >
              <svg
                class="icon icon__close"
                viewBox="0 0 365.696 365.696"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
                />
              </svg>
            </div>
          </div>
          <div
            class="modal__main"
          >
            <div
              class="w-100"
            >
              <div
                class="details"
              >
                <div
                  class="container "
                >
                  <img
                    alt="gutierrezbrian"
                    class="avatar"
                    src="https://dev.gambilife.com/ava/ano.svg"
                  />
                  <span
                    class="text--bold username"
                  >
                    gutierrezbrian
                  </span>
                </div>
                <ul
                  class="list"
                >
                  <li
                    class="list__item"
                  >
                    <div
                      class="list__item__label"
                    >
                      Date/Time
                    </div>
                    <div
                      class="list__item__value"
                    >
                      1591032136876
                    </div>
                  </li>
                  <li
                    class="list__item"
                  >
                    <div
                      class="list__item__label"
                    >
                      Bet Id
                    </div>
                    <div
                      class="list__item__value"
                    >
                      278192
                    </div>
                  </li>
                  <li
                    class="list__item"
                  >
                    <div
                      class="list__item__label"
                    >
                      Game
                    </div>
                    <div
                      class="list__item__value"
                    >
                      <svg
                        class="icon"
                        viewBox="0 0 35 35"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <defs>
                          <style>
                            
      .cls-2{fill:#091b33}
                          </style>
                        </defs>
                        <g
                          data-name="Layer 2"
                          id="Layer_2"
                        >
                          <g
                            data-name="Layer 5"
                            id="Layer_5"
                          >
                            <circle
                              cx="17.5"
                              cy="17.5"
                              fill="#fff"
                              r="17.5"
                            />
                            <path
                              class="cls-2"
                              d="M19.43 13.66l-5.18 3.12-3.12-5.17 5.18-3.12 3.12 5.17z"
                            />
                            <path
                              class="cls-2"
                              d="M19.43 13.66l-5.18 3.12-3.12-5.17 5.18-3.12 3.12 5.17z"
                            />
                            <path
                              class="cls-2"
                              d="M13.24 10.34l1-.61a6 6 0 00-6.2-2.85.53.53 0 00-.46.5.53.53 0 00.58.54 5 5 0 015.08 2.42z"
                            />
                            <path
                              class="cls-2"
                              d="M12.74 3.49L9.85 5.65 9 1l-.76 4.68-3.92-4.39 2.71 5-4-1.46 3.29 2.52L0 9.05l6.04-.03-2.64 3.51L7.38 9.9l1.05 3.45.82-4.19 2.21 1.54-1.3-3.3 2.58-3.91z"
                            />
                            <ellipse
                              class="cls-2"
                              cx="20.84"
                              cy="21.86"
                              rx="11.16"
                              ry="11.14"
                            />
                            <path
                              d="M29.58 19.39a11.13 11.13 0 00-5.4-5.84"
                              fill="none"
                              stroke="#fff"
                              stroke-miterlimit="10"
                              stroke-width="1.07"
                            />
                          </g>
                        </g>
                      </svg>
                      MINES
                    </div>
                  </li>
                  <li
                    class="list__item"
                  >
                    <div
                      class="list__item__label"
                    >
                      Bet
                    </div>
                    <div
                      class="list__item__value"
                    >
                      <div
                        class=""
                      >
                        <svg
                          class="icon"
                          viewBox="0 0 64 64"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            transform="translate(0.00630876,-0.00301984)"
                          >
                            <path
                              d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                            />
                            <path
                              class="icon__inner"
                              d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                            />
                          </g>
                        </svg>
                        <span>
                          0.00009425
                        </span>
                      </div>
                    </div>
                  </li>
                  <li
                    class="list__item"
                  >
                    <div
                      class="list__item__label"
                    >
                      Profit (x1.3026315789473684)
                    </div>
                    <div
                      class="list__item__value"
                    >
                      <div
                        class="text--positive"
                      >
                        <svg
                          class="icon"
                          viewBox="0 0 64 64"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            transform="translate(0.00630876,-0.00301984)"
                          >
                            <path
                              d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                            />
                            <path
                              class="icon__inner"
                              d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                            />
                          </g>
                        </svg>
                        <span>
                          +0.00000003
                        </span>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="modal__footer"
          >
            <div
              class="container"
            >
              <div>
                <button
                  class="button"
                >
                  <svg
                    class="arrow arrow--left"
                    viewBox="0 0 12 26"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style>
                        .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                      </style>
                    </defs>
                    <g
                      data-name="Layer 2"
                      id="Layer_2"
                    >
                      <g
                        data-name="Layer 1"
                        id="Layer_1-2"
                      >
                        <g
                          data-name="Group 197"
                          id="Group_197"
                        >
                          <path
                            class="cls-1"
                            d="M1.5 24.5l9-11"
                            data-name="Line 23"
                            id="Line_23"
                          />
                          <path
                            class="cls-1"
                            d="M1.5 1.5l9 12"
                            data-name="Line 24"
                            id="Line_24"
                          />
                        </g>
                      </g>
                    </g>
                  </svg>
                </button>
              </div>
              <div
                class="dots"
              >
                <div
                  class="dot__wrapper dot__wrapper--active"
                >
                  <div
                    class="dot"
                  />
                </div>
              </div>
              <div>
                <button
                  class="button"
                >
                  <svg
                    class="arrow arrow--right"
                    viewBox="0 0 12 26"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style>
                        .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                      </style>
                    </defs>
                    <g
                      data-name="Layer 2"
                      id="Layer_2"
                    >
                      <g
                        data-name="Layer 1"
                        id="Layer_1-2"
                      >
                        <g
                          data-name="Group 197"
                          id="Group_197"
                        >
                          <path
                            class="cls-1"
                            d="M1.5 24.5l9-11"
                            data-name="Line 23"
                            id="Line_23"
                          />
                          <path
                            class="cls-1"
                            d="M1.5 1.5l9 12"
                            data-name="Line 24"
                            id="Line_24"
                          />
                        </g>
                      </g>
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        aria-modal="true"
        class="modal scale--enter scale--enter-active"
        role="dialog"
      >
        <div
          class="modal__header"
        >
          <div
            class="modal__header__headline"
          >
            Bet Details
          </div>
          <div
            class="modal__header__close"
          >
            <svg
              class="icon icon__close"
              viewBox="0 0 365.696 365.696"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
              />
            </svg>
          </div>
        </div>
        <div
          class="modal__main"
        >
          <div
            class="w-100"
          >
            <div
              class="details"
            >
              <div
                class="container "
              >
                <img
                  alt="gutierrezbrian"
                  class="avatar"
                  src="https://dev.gambilife.com/ava/ano.svg"
                />
                <span
                  class="text--bold username"
                >
                  gutierrezbrian
                </span>
              </div>
              <ul
                class="list"
              >
                <li
                  class="list__item"
                >
                  <div
                    class="list__item__label"
                  >
                    Date/Time
                  </div>
                  <div
                    class="list__item__value"
                  >
                    1591032136876
                  </div>
                </li>
                <li
                  class="list__item"
                >
                  <div
                    class="list__item__label"
                  >
                    Bet Id
                  </div>
                  <div
                    class="list__item__value"
                  >
                    278192
                  </div>
                </li>
                <li
                  class="list__item"
                >
                  <div
                    class="list__item__label"
                  >
                    Game
                  </div>
                  <div
                    class="list__item__value"
                  >
                    <svg
                      class="icon"
                      viewBox="0 0 35 35"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <defs>
                        <style>
                          
      .cls-2{fill:#091b33}
                        </style>
                      </defs>
                      <g
                        data-name="Layer 2"
                        id="Layer_2"
                      >
                        <g
                          data-name="Layer 5"
                          id="Layer_5"
                        >
                          <circle
                            cx="17.5"
                            cy="17.5"
                            fill="#fff"
                            r="17.5"
                          />
                          <path
                            class="cls-2"
                            d="M19.43 13.66l-5.18 3.12-3.12-5.17 5.18-3.12 3.12 5.17z"
                          />
                          <path
                            class="cls-2"
                            d="M19.43 13.66l-5.18 3.12-3.12-5.17 5.18-3.12 3.12 5.17z"
                          />
                          <path
                            class="cls-2"
                            d="M13.24 10.34l1-.61a6 6 0 00-6.2-2.85.53.53 0 00-.46.5.53.53 0 00.58.54 5 5 0 015.08 2.42z"
                          />
                          <path
                            class="cls-2"
                            d="M12.74 3.49L9.85 5.65 9 1l-.76 4.68-3.92-4.39 2.71 5-4-1.46 3.29 2.52L0 9.05l6.04-.03-2.64 3.51L7.38 9.9l1.05 3.45.82-4.19 2.21 1.54-1.3-3.3 2.58-3.91z"
                          />
                          <ellipse
                            class="cls-2"
                            cx="20.84"
                            cy="21.86"
                            rx="11.16"
                            ry="11.14"
                          />
                          <path
                            d="M29.58 19.39a11.13 11.13 0 00-5.4-5.84"
                            fill="none"
                            stroke="#fff"
                            stroke-miterlimit="10"
                            stroke-width="1.07"
                          />
                        </g>
                      </g>
                    </svg>
                    MINES
                  </div>
                </li>
                <li
                  class="list__item"
                >
                  <div
                    class="list__item__label"
                  >
                    Bet
                  </div>
                  <div
                    class="list__item__value"
                  >
                    <div
                      class=""
                    >
                      <svg
                        class="icon"
                        viewBox="0 0 64 64"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          transform="translate(0.00630876,-0.00301984)"
                        >
                          <path
                            d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                          />
                          <path
                            class="icon__inner"
                            d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                          />
                        </g>
                      </svg>
                      <span>
                        0.00009425
                      </span>
                    </div>
                  </div>
                </li>
                <li
                  class="list__item"
                >
                  <div
                    class="list__item__label"
                  >
                    Profit (x1.3026315789473684)
                  </div>
                  <div
                    class="list__item__value"
                  >
                    <div
                      class="text--positive"
                    >
                      <svg
                        class="icon"
                        viewBox="0 0 64 64"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          transform="translate(0.00630876,-0.00301984)"
                        >
                          <path
                            d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                          />
                          <path
                            class="icon__inner"
                            d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                          />
                        </g>
                      </svg>
                      <span>
                        +0.00000003
                      </span>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="modal__footer"
        >
          <div
            class="container"
          >
            <div>
              <button
                class="button"
              >
                <svg
                  class="arrow arrow--left"
                  viewBox="0 0 12 26"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <style>
                      .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                    </style>
                  </defs>
                  <g
                    data-name="Layer 2"
                    id="Layer_2"
                  >
                    <g
                      data-name="Layer 1"
                      id="Layer_1-2"
                    >
                      <g
                        data-name="Group 197"
                        id="Group_197"
                      >
                        <path
                          class="cls-1"
                          d="M1.5 24.5l9-11"
                          data-name="Line 23"
                          id="Line_23"
                        />
                        <path
                          class="cls-1"
                          d="M1.5 1.5l9 12"
                          data-name="Line 24"
                          id="Line_24"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
              </button>
            </div>
            <div
              class="dots"
            >
              <div
                class="dot__wrapper dot__wrapper--active"
              >
                <div
                  class="dot"
                />
              </div>
            </div>
            <div>
              <button
                class="button"
              >
                <svg
                  class="arrow arrow--right"
                  viewBox="0 0 12 26"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <style>
                      .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                    </style>
                  </defs>
                  <g
                    data-name="Layer 2"
                    id="Layer_2"
                  >
                    <g
                      data-name="Layer 1"
                      id="Layer_1-2"
                    >
                      <g
                        data-name="Group 197"
                        id="Group_197"
                      >
                        <path
                          class="cls-1"
                          d="M1.5 24.5l9-11"
                          data-name="Line 23"
                          id="Line_23"
                        />
                        <path
                          class="cls-1"
                          d="M1.5 1.5l9 12"
                          data-name="Line 24"
                          id="Line_24"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
